meta-resin-common/recipes-support/resin-expand/resin-filesystem-expand/resin-filesystem-expand.service

```diff
@@ -1,10 +1,11 @@
 [Unit]
 Description=Expand resin-data partition
 DefaultDependencies=no
-Requires=dev-disk-by\x2dlabel-resin\x2ddata.device
-After=dev-disk-by\x2dlabel-resin\x2ddata.device
-Before=mnt-data.mount umount.target
+Requires=systemd-udev-settle.service
+After=systemd-udev-settle.service
+Before=mnt-data.service umount.target
 Conflicts=umount.target
+ConditionVirtualization=!docker
```

Comment:
If the condition is in place, changing the requires based on the container argument should not be necessary, as this service will not run in a container. Is there another reason to move to udev-settle as a dependency?

Reply:
Correct but the requirements are added for the case where you run this service on a board. You need to way for udev to settle to make sure you have the labels created.

Reply:
But this service requires resin-data, I do not understand why we moved from the specific requires to the broader one with udev-settle.

Reply:
@telphan This is because in container there are no device units activated/plugged. So this service needs to run in a container without waiting for device units which never get plugged (due to the issue documented in various commits here). This is because we can't depend on the device unit anymore. But when running on actual devices we need to make sure that the resin-data is actually available so we depend on udev settle.